share/completions/magento.fish

```diff
@@ -0,0 +1,720 @@
+__fish_magento_register_command maintenance:enable -d "Enables maintenance mode";
+__fish_magento_register_command maintenance:status -d "Displays maintenance mode status";
+
+__fish_magento_register_command module:disable -d "Disables specified modules";
+__fish_magento_register_command module:enable -d "Enables specified modules";
+__fish_magento_register_command module:status -d "Displays status of modules";
+__fish_magento_register_command module:uninstall -d "Uninstalls modules installed by composer";
+
+__fish_magento_register_command sampledata:deploy -d "Deploy sample data modules";
+__fish_magento_register_command sampledata:remove -d "Remove all sample data packages from composer.json";
+__fish_magento_register_command sampledata:reset -d "Reset all sample data modules for re-installation";
+
+__fish_magento_register_command setup:backup -d "Takes backup of Magento Application code base, media and database";
+__fish_magento_register_command setup:config:set -d "Creates or modifies the deployment configuration";
+__fish_magento_register_command setup:cron:run -d "Runs cron job scheduled for setup application";
+__fish_magento_register_command setup:db-data:upgrade -d "Installs and upgrades data in the DB";
+__fish_magento_register_command setup:db-schema:upgrade -d "Installs and upgrades the DB schema";
+__fish_magento_register_command setup:db:status -d "Checks if DB schema or data requires upgrade";
+__fish_magento_register_command setup:di:compile -d "Generates DI configuration and all missing classes that can be auto-generated";
+__fish_magento_register_command setup:install -d "Installs the Magento application";
+__fish_magento_register_command setup:performance:generate-fixtures -d "Generates fixtures";
+__fish_magento_register_command setup:rollback -d "Rolls back Magento Application codebase, media and database";
+__fish_magento_register_command setup:static-content:deploy -d "Deploys static view files";
+__fish_magento_register_command setup:store-config:set -d "Installs the store configuration";
+__fish_magento_register_command setup:uninstall -d "Uninstalls the Magento application";
+__fish_magento_register_command setup:upgrade -d "Upgrades the Magento application, DB data, and schema";
+
+__fish_magento_register_command theme:uninstall -d "Uninstalls a theme";
+
+#
+# help
+#
+__fish_magento_register_command_option help -x -a "(__fish_print_magento_commands_list)" -d "Displays help for a command";
+
+#
+# list
+#
+__fish_magento_register_command_option list -f -l xml -d "To output list as XML";
+__fish_magento_register_command_option list -f -l raw -d "To output raw command list";
+__fish_magento_register_command_option list -f -l format -a "(__fish_print_magento_list_formats)" -d "To output list in other formats (default: txt)";
+
+#
+# admin:user:create
+#
+__fish_magento_register_command_option admin:user:create -f -r -l admin-user          -d "(Required) Admin user";
```

Comment:
Since these are required, it would be great if we could just automatically suggest them. That currently doesn't happen with options, so you'll have to suggest one via "-a".

So that would be

```fish
# Force suggesting options
__fish_magento_register_command_option admin:user:create -a '--admin-user'
```